redash/handlers/api.py

```diff
@@ -92,6 +92,7 @@ def json_representation(data, code, headers=None):
 api.add_org_resource(QueryResultResource,
                      '/api/query_results/<query_result_id>.<filetype>',
                      '/api/query_results/<query_result_id>',
+                     '/api/queries/<query_id>/results',
                      '/api/queries/<query_id>/results.<filetype>',
                      '/api/queries/<query_id>/results/<query_result_id>.<filetype>',
                      endpoint='query_result')
```

Comment:
It bothers me that we have so many different URLs on a single handler. One side effect of this, is now someone can call `POST /api/queries/<query id>/results.json` or `POST /api/queries/<query id>/results/<query result id>.json` and generate a server error (500) instead of receiving a method not supported error.

Let's start splitting them by adding this new one to its own handler?

Reply:
You're right, it bothered me while adding the endpoint. I just couldn't find a proper definition for such handler.

Reply:
I just realized that we're limited by what Flask-Restful allows. This is the case where if we had simple function based Flask handlers it would probably be less limited. :|